### ISSUE TITLE
[REQUEST] Option to validate the model more frequently than every epoch

### ISSUE THREAD

**Issue by shamshaw** [neuralhydrology/neuralhydrology#131]
[Feature request] -- It would be nice to have an option to log validation loss more frequently than every epoch during training. 

I have seen on a few model trainings runs that I have done (particularly with large number of gages and long training periods), the model appears to fit essentially just during the first epoch. It would be nice to have an option to do validation (at least the loss, metrics would not be as important) more frequently than every epoch. I think this could be accomplished with an option to validate after number of batches, or an approximate alternative could be to only use a % of the data in one "epoch"

**Comment by kratzert** [comment]
Hi,

sure, we can add that. I actually just did this recently in some different code and it requires minimal changes/additions. I will push this with an upcoming version bump.

In the meantime, if you already want to do that, just change the for-loop here https://github.com/neuralhydrology/neuralhydrology/blob/master/neuralhydrology/training/basetrainer.py#L283

to be:

```python
for i, data in enumerate(pbar):
    if self._max_iter_per_epoch and if i > self._max_iter_per_epoch:
        break
´´´

and in the init of the trainer class you set self._max_iter_per_epoch to `None` if you want to train on the full epoch or to some integer, if you only want to train for some max number of update steps (after which NH would consider this a full epoch and could run validation).

**Comment by shamshaw** [comment]
@kratzert Great! Appreciate the quick suggestion on modifying to code in the meantime.

p.s. Excited to see all the forecasting models added in the latest release - looking forward to trying those out. So, thanks to the neural-hydrology team for the effort!

**Comment by kratzert** [comment]
closed by #143
